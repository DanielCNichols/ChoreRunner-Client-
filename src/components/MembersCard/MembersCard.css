.memberCard {
  position: relative;
  width: 80%;
  min-width: 300px;
  min-height: 250px;
  margin: 10px auto 20px auto;
  /* font-family: 'Fredoka One', cursive;
    font-size: 23px; */
  background-color: #3e5ad6;
  -webkit-border-radius: 7px;
  -moz-border-radius: 7px;
  -o-border-radius: 7px;
  border-radius: 9px;
  border: #334ab0 solid 1px;
  box-shadow: 0 0.3em #334ab0;
  display: grid;
  box-sizing: border-box;
  grid-template-columns: auto auto;
  grid-template-rows: auto auto auto auto;
  /* padding: 20px 10px; */
  border: 3px solid white;
  border-radius: 10px;
  min-width: 310px;
  font-family: 'Electrolize', sans-serif;
  color: white;
}

.member-rule {
    grid-column-start: 1;
    grid-column-end: 3;
    grid-row-start: 2;
    width: 100%;
    height: 2px;
    background-color: white;
    margin: 0;
}

.memberCard h3 {
  grid-column-start: 1;
  grid-row-start: 1;
  margin: 0;
  padding: 10px;
  font-family: 'Press Start 2P', cursive;
  font-size: 1.2rem;
  color: #ffd900;
  text-align: left;
  display: flex;
  align-items: center;
}

.memberCard h4 {
    margin: 0;
    text-align: center;
    font-family: "Press Start 2p", cursive;
    color: #ffd900;
    font-size: 15px;
}

.score {
  grid-column-start: 2;
  grid-row-start: 1;
  margin: 0;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  padding: 10px;
}

.task-container {
    grid-column-start: span 2;
    grid-row-start: 3;
    width: 100%;

}

.memberCard-button-container {
    grid-column-start: span 2;
    grid-row-start: 4;
    padding: 10px;
    display: flex;
    flex-direction: row;
    justify-content: center;
}

.memberCard ul {
    padding: 0;
    list-style : none;
}

.memberCard-button-container button {
    text-align: center;
  margin: 15px auto;
  font-family: 'Electrolize', sans-serif;
  font-size: 1em;
  padding: 15px;
  background-color: rgb(255, 217, 0);
  text-shadow: -1px -1px rgb(105, 90, 0), 1px 1px white;
  color: #725f00;
  -webkit-border-radius: 7px;
  -moz-border-radius: 7px;
  -o-border-radius: 7px;
  border-radius: 7px;
  box-shadow: 0 0.3em rgb(134, 112, 0);
  cursor: pointer;
  min-width: 123px;
  max-width: 123px;
}