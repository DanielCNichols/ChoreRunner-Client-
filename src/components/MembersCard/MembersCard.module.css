@import '../../vars.css';

.memberCard {
  width: 100%;
  min-width: 300px;
  margin-bottom: 20px;
  background-color: var(--color-menus);
  -webkit-border-radius: 6px;
  -moz-border-radius: 6px;
  -o-border-radius: 6px;
  border-radius: 6px;
  border: 2px solid white;
  font-family: var(--font-body);
  color: white;
}

.memberCard header {
  display: flex;
  width: 100%;
  padding: 10px;
  justify-content: space-between;
  align-items: center;
  border-bottom: 2px solid white;
  margin-bottom: 20px;
}

.progressBarContainer {
  display: flex;
  justify-content: center;
  width: 30%;
}

.memberCard header h3,
.memberCard header p {
  padding: 0;
  margin: 0;
}

.memberCard header h3 {
  font-size: 4vw;
  max-width: 130px;
  text-align: left;
}

.cardBody {
  padding: 0 10px;
}

.cardBody p {
  text-align: left;
}

.memberCard h4 {
  margin: 10px auto;
  text-align: left;
  color: white;
  font-size: 4vw;
}
.memberCard ul {
  padding: 0;
  width: 100%;
}
