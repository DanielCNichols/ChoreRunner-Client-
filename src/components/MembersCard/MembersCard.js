import React from 'react';
import Modal from '../Modal/Modal';
import EditMember from '../EditMember/EditMember';
import Task from '../Task/Task';
import './MembersCard.css';

export default class MembersCard extends React.Component {
  state = {
    assignedTasks: [],
    completedTasks: [],
    editingMember: false,
    error: null,
  };

  componentDidMount() {
    this.arrangeTasks();
  }

  arrangeTasks() {
    let { tasks } = this.props.member;
    let assigned = [];
    let completed = [];

    tasks.map(task => {
      return task.status === 'assigned'
        ? assigned.push(task)
        : completed.push(task);
    });

    this.setState({ assignedTasks: assigned, completedTasks: completed });
  }

  toggleTaskForm = () => {
    this.setState({ editingTask: !this.state.editingTask });
  };

  toggleMemberForm = () => {
    this.setState({ editingMember: !this.state.editingMember });
  };

  render() {
    const {
      member,
      householdId,
      editMember,
      editTask,
      tasks,
      approveTask,
      rejectTask,
      deleteMember,
      deleteTask,
    } = this.props;

    const { editingMember, assignedTasks, completedTasks } = this.state;

    return (
      <div className="memberCard">
        <h3>{member.name}</h3>
        <p className="score">Score: {member.total_score}</p>

        {editingMember ? (
            <Modal>
            <EditMember
              householdId={householdId}
              editMember={editMember}
              handleCancel={this.toggleMemberForm}
              member={member}
            ></EditMember>
          </Modal>
        ) : null}
        {/* Put each task as its own component.  */}
        <div className="task-container">
        <hr className="member-rule"></hr>
          <ul>
            <h4>Assigned Tasks</h4>
            {assignedTasks.length ? 
            tasks.map(task => {
              return task.status === 'assigned' ? (
                <Task
                  deleteTask={deleteTask}
                  editTask={editTask}
                  key={task.id}
                  householdId={householdId}
                  task={task}
                />
              ) : null;
            }) : <li><p>No tasks assigned</p></li>}
          </ul>
          <ul>
            <h4>Completed</h4>
            {completedTasks.length ? 
            tasks.map(task => {
              return task.status === 'completed' ? (
                <Task
                  key={task.id}
                  rejectTask={rejectTask}
                  approveTask={approveTask}
                  householdId={householdId}
                  task={task}
                />
              ) : null;
            }) : <li><p>No tasks completed</p></li>}
          </ul>
        </div>
        <div className="memberCard-button-container">
          <button onClick={this.toggleMemberForm}>Edit</button>
          <button
            onClick={() => {
              deleteMember(member.id);
            }}
          >
            Delete
          </button>
        </div>
      </div>
    );
  }
}
