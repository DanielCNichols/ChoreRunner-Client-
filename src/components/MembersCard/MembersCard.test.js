import React from 'react';
import ReactDOM from 'react-dom';
import MembersCard from './MembersCard';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('MembersCard component testing', () => {
  const member = {
    name: 'daniel',
    level_id: 1,
    pointsToNextLevel: 2,
    completedTasks: [],
    assignedTasks: [],
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <MemoryRouter>
        <MembersCard member={member} />
      </MemoryRouter>,
      div
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <MembersCard member={member} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
