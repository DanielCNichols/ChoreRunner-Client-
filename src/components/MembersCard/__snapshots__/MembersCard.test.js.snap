// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MembersCard component testing renders UI as expected 1`] = `
<div
  className="memberCard"
>
  <header>
    <h3>
      daniel
    </h3>
    <p>
      level: 
      1
    </p>
    <div
      className="progressBarContainer"
    >
      <div
        style={
          Object {
            "backgroundColor": "blue",
            "border": "2px solid white",
            "height": "30px",
            "position": "relative",
            "width": "100%",
          }
        }
      >
        <span
          style={
            Object {
              "color": "#ffd900",
              "display": "inline-block",
              "fontWeight": "bold",
              "padding": "5px",
              "position": "relative",
              "textAlign": "center",
              "verticalAlign": "middle",
              "width": "100%",
              "zIndex": "2",
            }
          }
        >
          8 / 10
        </span>
        <div
          style={
            Object {
              "backgroundColor": "white",
              "height": "100%",
              "position": "absolute",
              "textAlign": "right",
              "top": "0",
              "width": "80%",
            }
          }
        />
      </div>
    </div>
  </header>
  <div
    className="cardBody"
  >
    <h4
      className="videoGameTitles"
    >
      Completed Tasks
    </h4>
    <ul
      className="completed"
    />
    <p>
      daniel
       has not completed any tasks... yet!
    </p>
    <h4
      className="videoGameTitles"
    >
      Assigned Tasks
    </h4>
    <ul
      className="assigned"
    />
    <p>
      daniel
       has no tasks.
    </p>
  </div>
</div>
`;
